src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java

```diff
@@ -188,17 +239,40 @@ private boolean verifyCodesign(Path target) throws IOException, InterruptedExcep
 
     private Path getEntitlementsPath() throws IOException {
         Path entitlements = rootPath.resolve("Entitlements.plist");
-        if (Files.exists(entitlements)) {
-            return entitlements;
+        boolean macAppStore = projectConfiguration.getReleaseConfiguration().isMacAppStore();
+        if (!macAppStore) {
+            return Files.exists(entitlements) ? entitlements : null;
+        }
+
+        Path tmpEntitlements = paths.getTmpPath().resolve("Entitlements.plist");
+        if (!Files.exists(entitlements)) {
+            entitlements = FileOps.copyResource("/native/macosx/assets/Entitlements.plist", tmpEntitlements);
+        }
+        try {
+            NSDictionaryEx dict = new NSDictionaryEx(entitlements);
+            SANDBOX_KEYS.stream()
+                    .filter(key -> dict.get(key) == null)
+                    .forEach(key -> dict.put(key, true));
+
+            if (provisionProfile != null) {
+                NSDictionaryEx profileEntitlements = provisionProfile.getEntitlements();
+                Arrays.stream(profileEntitlements.getAllKeys())
+                        .filter(key -> dict.get(key) == null)
+                        .forEach(key -> dict.put(key, profileEntitlements.get(key)));
+            }
+
+            dict.saveAsXML(tmpEntitlements);
+        } catch (ParserConfigurationException | ParseException | SAXException | PropertyListFormatException e) {
+            e.printStackTrace();
```

Comment:
probably don't ignore exceptions here. you want to fail the build of this happens?

Reply:
Makes sense, since the user can add a malformed plist file.